### PR TITLE
BUG-CoW: DataFrame constructed from Series not respecting CoW

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

right now we back the DataFrame by a read-only array, before the read-only pr this did not respect CoW.

Follow-up will be to do the same for Index.

### CODE REVIEW

File: pandas/tests/copy_view/test_constructors.py [hunk 1]
```diff
@@ -164,6 +165,38 @@ def test_dataframe_from_dict_of_series_with_reindex(dtype):
...
+    ser = Series([1, 2], dtype="int64")
+    df = DataFrame(ser, dtype="int32")
+    assert not np.shares_memory(get_array(ser), get_array(df, 0))
+    if using_copy_on_write:
+        assert df._mgr._has_no_reference(0)
+
+
+def test_dataframe_from_series_infer_datetime(using_copy_on_write):
+    ser = Series([Timestamp("2019-12-31"), Timestamp("2020-12-31")], dtype=object)
+    df = DataFrame(ser)
```
Comment (jorisvandenbossche) [inline pandas/tests/copy_view/test_constructors.py:194]: We should stop inferring this, but that's another issue ;)

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: pandas/tests/copy_view/test_constructors.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/copy_view/test_constructors.py:194]: 😅

**Comment by phofl** [comment]
I'll merge this to get it into 2.0 and to make the follow up

**Pull request merged**